plugins/turing/runner/experiment_runner.go

```diff
@@ -78,12 +82,65 @@ func (er *experimentRunner) GetTreatmentForRequest(
 				projectId,
 				statusCode,
 				err,
-				api.FetchTreatmentRequestBody{AdditionalProperties: requestParams},
+				filterParams,
 				requestFilter,
 			)
 		}
 	}()
 
+	var lookupRequestFilters []models.SegmentFilter
+	var errorLog *monitoring.ErrorResponseLog
+
+	if er.appContext.AssignedTreatmentLogger != nil {
```

Comment:
This block is almost a perfect copy-paste from https://github.com/caraml-dev/xp/blob/f5eb2bd3c3ce301f392a1120232748a9255ab998/treatment-service/controller/treatment.go#L67.